src/arrayMethodSort.js

```diff
@@ -1,12 +1,78 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const undefinedArr = [];
+
+    for (let i = 0; i < this.length; ) {
+      if (this[i] === null) {
+        undefinedArr.unshift(...this.splice(i, 1));
+      }
+
+      if (typeof this[i] === 'undefined') {
+        undefinedArr.push(...this.splice(i, 1));
+      } else {
+        this[i] = this[i].toString();
+        i++;
+      }
+    }
+
+    for (let j = 0; j < this.length; j++) {
+      for (let i = 0; i < this.length; i++) {
+        let result;
+
+        if (compareFunction) {
+          result = compareFunction(this[i], this[i + 1]);
+        }
+
+        if (!compareFunction) {
+          result = this[i] > this[i + 1];
```

Comment:
The comparison logic here is incorrect. The expression `this[i] > this[i + 1]` will return a boolean, not a number. This should be adjusted to return a positive or negative number for sorting purposes.